File: DockerSpringBootAngular/PetClinicAngularRouting/src/app/pet.ts
import { Owner } from "./owner";

export class Pet {

    id: number;
    ownerId: number;
    name: String;
    type: String;
    dob: String;
    breed: String;
    owner: Owner;

    constructor(id: number, ownerId: number, name: String, type: String, dob: String, breed: String, owner: Owner) {
        this.id = id;
        this.ownerId = ownerId;
        this.name = name;
        this.type = type;
        this.dob = dob;
        this.breed = breed;
        this.owner = owner;
    }
}

File: DockerSpringBootAngular/PetClinicSpringBootRESTfulService/src/main/java/com/pet/clinic/bean/Pet.java
package com.pet.clinic.bean;

import org.springframework.context.annotation.Scope;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

@Entity
@Scope("prototype")
public class Pet {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	private String name;
	private String type;
	private String breed;
	private String dob;
	
	@ManyToOne
	private Owner owner;
	
	public Pet() {

	}
	public Pet(Integer id, String name, String type, String breed, String dob, Owner owner) {
		this.id = id;
		this.name = name;
		this.type = type;
		this.breed = breed;
		this.dob = dob;
		this.owner = owner;
	}
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getType() {
		return type;
	}
	public void setType(String type) {
		this.type = type;
	}
	public String getBreed() {
		return breed;
	}
	public void setBreed(String breed) {
		this.breed = breed;
	}
	public String getDob() {
		return dob;
	}
	public void setDob(String dob) {
		this.dob = dob;
	}
	public Owner getOwner() {
		return owner;
	}
	public void setOwner(Owner owner) {
		this.owner = owner;
	}
}
